libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,130 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solAssert(_code.isArray());
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const nextItem = next(current);
+		if (nextItem != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			nextItem->type() == AssemblyItemType::Operation &&
+			nextItem->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(nextItem);
+	}
+
+	for (auto const& item: m_items)
+		if (item.type() == AssemblyItemType::Operation && item.instruction() == Instruction::JUMPDEST)
+			throw langutil::Error(
+				1285_error,
+				langutil::Error::Type::FatalError,
+				"JUMPDEST instruction found that was not followed by tag."
+			);
+}
+
+AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json)
+{
+	solAssert(isOfType<string>(_json["name"]));
+	solAssert(isOfType<int>(_json["begin"]));
+	solAssert(isOfType<int>(_json["end"]));
+	solAssert(isOfType<int>(_json["source"]));
+	solAssert(isOfTypeIfExists<string>(_json, "value"));
+	solAssert(isOfTypeIfExists<int>(_json, "modifierDepth"));
+	solAssert(isOfTypeIfExists<string>(_json, "jumpType"));
```

Comment:
Good point. I just introduced a new macro `solThrowIfNot` that only throws, if a condition was not met. Maybe the name is not the best.

Reply:
Ok, I think its better to have a `solThrowIf`.

Reply:
Why not just use an `if`? You'll still want to put the error message in there so it will likely not fit on a single line anyway.

Reply:
If you really want to keep it, maybe we should call it `solRequire()` by analogy to `assert()`/`require()` in Solidity. It's then also more consistent with `solAssert()` because the condition is not negated.

Reply:
I decided to keep it and I renamed that to `solRequire`.